cmd/kubeadm/app/cmd/upgrade/diff_test.go

```diff
@@ -124,3 +136,73 @@ func TestRunDiff(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateManifests(t *testing.T) {
+	rand.Seed((time.Now().UnixNano()))
+	// Create valid manifest path
+	apiServerManifest, err := createTestRunDiffFile([]byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(apiServerManifest)
+	controllerManagerManifest, err := createTestRunDiffFile([]byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(controllerManagerManifest)
+	schedulerManifest, err := createTestRunDiffFile([]byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(schedulerManifest)
+	// Create a directory path that does not exist
+	notExistDirPath := "/xxxxxxxxxx"
+	_, err = os.Stat(notExistDirPath)
+	for !os.IsNotExist(err) {
+		notExistDirPath = "/" + createRandString(10)
+		_, err = os.Stat(notExistDirPath)
+	}
+	// Create a file path that does not exist
+	notExistFilePath := notExistDirPath + "/xxxxx.yaml"
```

Comment:
while your solution is not wrong you can just use this:
```
notExistFilePath := `./foobar123456`
```

we don't expect this file to exist.